components/suggestion/switch_channel_provider.jsx

```diff
@@ -581,19 +610,22 @@ export default class SwitchChannelProvider extends Provider {
                 continue;
             }
 
-            const wrappedChannel = this.userWrappedChannel(user);
-
-            const currentUserId = getCurrentUserId(getState());
-
             const channelName = Utils.getDirectChannelName(currentUserId, user.id);
-            const channel = getChannelByName(getState(), channelName);
+            const channel = getChannelByName(state, channelName);
+
+            const wrappedChannel = this.userWrappedChannel(user, channel);
 
             if (channel && members[channel.id]) {
                 wrappedChannel.last_viewed_at = members[channel.id].last_viewed_at;
             } else if (skipNotMember) {
                 continue;
             }
 
+            const unread = allUnreadChannelIdsSet.has(channel.id);
```

Comment:
@AshishDhama I think the failed tests in E2E were related here (see screenshot). Please run the tests in your local and inspect the console logs if you're getting the same error.
<img width="654" alt="Screen Shot 2022-04-28 at 7 09 39 PM" src="https://user-images.githubusercontent.com/5334504/165739672-392b39c3-9fc8-4c13-b089-3e38321fb918.png">
<img width="401" alt="Screen Shot 2022-04-28 at 7 09 50 PM" src="https://user-images.githubusercontent.com/5334504/165739678-2b1a470a-20c2-4568-9e3f-65e4c264f4d2.png">